python/ql/test/experimental/dataflow/coverage/test.py

```diff
@@ -39,93 +42,241 @@ def test_floatnumber_literal():
 
 def test_imagnumber_literal():
     x = 42j
-    SINK(x)
+    SINK(x) # Flow missing
 
 # 6.2.3. Parenthesized forms
 def test_parenthesized_form():
     x = (SOURCE)
-    SINK(x)
+    SINK(x) # Flow missing
 
 # 6.2.5. List displays
 def test_list_display():
     x = [SOURCE]
-    SINK(x[0])
+    SINK(x[0]) # Flow missing
 
 def test_list_comprehension():
     x = [SOURCE for y in [3]]
-    SINK(x[0])
+    SINK(x[0]) # Flow missing
 
 def test_nested_list_display():
     x = [* [SOURCE]]
-    SINK(x[0])
+    SINK(x[0]) # Flow missing
 
 # 6.2.6. Set displays
 def test_set_display():
     x = {SOURCE}
-    SINK(x.pop())
+    SINK(x.pop()) # Flow missing
 
 def test_set_comprehension():
     x = {SOURCE for y in [3]}
-    SINK(x.pop())
+    SINK(x.pop()) # Flow missing
 
 def test_nested_set_display():
     x = {* {SOURCE}}
-    SINK(x.pop())
+    SINK(x.pop()) # Flow missing
 
 # 6.2.7. Dictionary displays
 def test_dict_display():
     x = {"s": SOURCE}
-    SINK(x["s"])
+    SINK(x["s"]) # Flow missing
 
 def test_dict_comprehension():
     x = {y: SOURCE for y in ["s"]}
-    SINK(x["s"])
+    SINK(x["s"]) # Flow missing
 
 def test_nested_dict_display():
     x = {** {"s": SOURCE}}
-    SINK(x["s"])
+    SINK(x["s"]) # Flow missing
 
 # 6.2.8. Generator expressions
 def test_generator():
     x = (SOURCE for y in [3])
-    SINK([*x][0])
+    SINK([*x][0]) # Flow missing
 
-# List taken from https://docs.python.org/3/reference/expressions.html
-# 6. Expressions
-# 6.1. Arithmetic conversions
-# 6.2. Atoms
-# 6.2.1. Identifiers (Names)
-# 6.2.2. Literals
-# 6.2.3. Parenthesized forms
-# 6.2.4. Displays for lists, sets and dictionaries
-# 6.2.5. List displays
-# 6.2.6. Set displays
-# 6.2.7. Dictionary displays
-# 6.2.8. Generator expressions
 # 6.2.9. Yield expressions
+def gen(x):
+    yield x
+
+def test_yield():
+    g = gen(SOURCE)
+    SINK(next(g)) # Flow missing
+
+def gen_from(x):
+    yield from gen(x)
+
+def test_yield_from():
+    g = gen_from(SOURCE)
+    SINK(next(g)) # Flow missing
+
+# a statement rather than an expression, but related to generators
+def test_for():
+    for x in gen(SOURCE):
```

Comment:
Also test negative case `for x in gen(NONSOURCE)`

Reply:
Again, I think this is just testing `isSource` (and perhaps the underlying library) rather than our coverage of Python constructs.